src/AutoRest.CSharp/Mgmt/Decorator/ParameterMappingBuilder.cs

```diff
@@ -197,18 +195,18 @@ private static IEnumerable<List<Segment>> SplitDiffIntoPairs(RequestPath diff)
             return result;
         }
 
-        public static FormattableString GetValueExpression(CSharpType type, FormattableString rawExpression)
+        public static ValueExpression GetValueExpression(CSharpType type, ValueExpression valueExpression)
         {
             if (TypeFactory.IsStringLike(type))
-                return rawExpression;
+                return valueExpression;
 
             if (!type.IsFrameworkType)
             {
                 if (type.Implementation is EnumType enumType && !enumType.IsExtensible)
                 {
-                    return $"{rawExpression}.To{enumType.Declaration.Name}()";
+                    return new InstanceMethodCallExpression(valueExpression, $"To{enumType.Declaration.Name}", Array.Empty<ValueExpression>(), false);
```

Comment:
Could we also add the method for enum serialization and deserialization to the `EnumType` class? We are kind of constructing the names of these two methods in multiple places (this is one of the places) and assuming those names.

For this deserialization method, the name is kind of stable, but for serialization, it varies from the underlying type. Therefore I believe it should benefit if we put these names inside `EnumType`.

Reply:
Done.